In [1]:
import pandas as pd
import numpy as np

In [2]:
xls = pd.ExcelFile(r'D:\Gianfranco SICILIANO\reshape_JanMay2020.xlsx')

# Now you can list all sheets in the file
sheet_names=list(xls.sheet_names)
sheet_names

['General info from parsing',
 'Negative',
 'Positive',
 'Uncertainity',
 'Litigious',
 'Strong Modals',
 'Weak Modals',
 'Risk',
 'Covid',
 'Risk Financial',
 'Risk Other Idiosyncratic',
 'Risk Legal and Regulatory',
 'Risk Other Systematic']

In [7]:
dfex_new=pd.DataFrame()
for i in sheet_names[1:]:
    dfex_new[i]=pd.Series([j.title() for j in pd.read_excel(r'D:\Gianfranco SICILIANO\reshape_JanMay2020.xlsx', sheet_name=i).columns])

In [94]:
dfex_new.head(9)

Negative    Positive Uncertainity       Litigious Strong Modals  \
0        Isin        Isin         Isin            Isin          Isin   
1     Country     Country      Country         Country       Country   
2    Exchange    Exchange     Exchange        Exchange      Exchange   
3  Bvd_Sector  Bvd_Sector   Bvd_Sector      Bvd_Sector    Bvd_Sector   
4         Sic         Sic          Sic             Sic           Sic   
5     Comname     Comname      Comname         Comname       Comname   
6  Disclosure  Disclosure   Disclosure      Disclosure    Disclosure   
7        Date        Date         Date            Date          Date   
8     Abandon        Able     Abeyance  Abovementioned        Always   

  Weak Modals        Risk       Covid                         Risk Financial  \
0        Isin        Isin        Isin                                   Isin   
1     Country     Country     Country                                Country   
2    Exchange    Exchange    Exchange                               Exchange   
3  Bvd_Sector  Bvd_Sector  Bvd_Sector                             Bvd_Sector   
4         Sic         Sic         Sic                                    Sic   
5     Comname     Comname     Comname                                Comname   
6  Disclosure  Disclosure  Disclosure                             Disclosure   
7        Date        Date        Date                                   Date   
8      Almost      Affect       China  Anti-Takeover (Provisions|Provision)    

  Risk Other Idiosyncratic Risk Legal and Regulatory Risk Other Systematic  
0                     Isin                      Isin                  Isin  
1                  Country                   Country               Country  
2                 Exchange                  Exchange              Exchange  
3               Bvd_Sector                Bvd_Sector            Bvd_Sector  
4                      Sic                       Sic                   Sic  
5                  Comname                   Comname               Comname  
6               Disclosure                Disclosure            Disclosure  
7                     Date                      Date                  Date  
8             Acquisition          Adverse Judgment           Afghanistan

In [10]:
dfex_new.iloc[8:,:].to_csv("GF_clean_words2020.csv", index=False)  #saving all the words which starts from 8th row

In [4]:
import pandas as pd
dfn=pd.read_excel(r'D:\Gianfranco SICILIANO\reshape_JanMay2020.xlsx', sheet_name="General info from parsing")
dfn.head()

isin  country               exchange    bvd_sector   sic  comname  \
0  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
1  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
2  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
3  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
4  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   

  disclosure       date  words#  
0        pre 2020-03-10     NaN  
1       earn 2020-04-27     NaN  
2     other1 2020-01-31     NaN  
3     other2 2020-02-05     NaN  
4     other3 2020-04-02     NaN

In [5]:
dfn.shape

(12813, 9)

In [13]:
dfn=dfGP[["isin","disclosure"]]
dfn.head()

isin disclosure
0  AT0000609607        pre
1  AT0000609607       earn
2  AT0000609607     other1
3  AT0000609607     other2
4  AT0000609607     other3

In [6]:
dfn["File_name_output"]=dfn["isin"].str.strip()+"_"+dfn["disclosure"].str.strip()

In [7]:
dfn.head()

isin  country               exchange    bvd_sector   sic  comname  \
0  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
1  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
2  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
3  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
4  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   

  disclosure       date  words#     File_name_output  
0        pre 2020-03-10     NaN     AT0000609607_pre  
1       earn 2020-04-27     NaN    AT0000609607_earn  
2     other1 2020-01-31     NaN  AT0000609607_other1  
3     other2 2020-02-05     NaN  AT0000609607_other2  
4     other3 2020-04-02     NaN  AT0000609607_other3

In [8]:
dfn["File_name"]=dfn["File_name_output"].str.strip()

In [9]:
dfn.head()

isin  country               exchange    bvd_sector   sic  comname  \
0  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
1  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
2  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
3  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
4  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   

  disclosure       date  words#     File_name_output            File_name  
0        pre 2020-03-10     NaN     AT0000609607_pre     AT0000609607_pre  
1       earn 2020-04-27     NaN    AT0000609607_earn    AT0000609607_earn  
2     other1 2020-01-31     NaN  AT0000609607_other1  AT0000609607_other1  
3     other2 2020-02-05     NaN  AT0000609607_other2  AT0000609607_other2  
4     other3 2020-04-02     NaN  AT0000609607_other3  AT0000609607_other3

In [10]:
dfn.drop(["words#"],axis=1,inplace=True)  
dfn.drop(["File_name_output"],axis=1,inplace=True)  

In [11]:
dfn.head()

isin  country               exchange    bvd_sector   sic  comname  \
0  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
1  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
2  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
3  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
4  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   

  disclosure       date            File_name  
0        pre 2020-03-10     AT0000609607_pre  
1       earn 2020-04-27    AT0000609607_earn  
2     other1 2020-01-31  AT0000609607_other1  
3     other2 2020-02-05  AT0000609607_other2  
4     other3 2020-04-02  AT0000609607_other3

In [12]:
dfn.shape

(12813, 9)

In [18]:
dfn.to_csv("File_name_output2020.csv",index=False)

In [13]:
dfl=pd.read_csv("Complete_GF_data_2020.csv")
dfl.head()

File_name                                               Data
0  AT0000609607_2020_earn  Press release\nPORR: Record order backlog prov...
1   AT0000609607_2020_pre  Press release\nPreliminary figures 2019:\nPORR...
2     AT0000609607_other1  Press Release\nPORR starts into 2020 with succ...
3     AT0000609607_other2  Press release\nPORR to modernise strategically...
4     AT0000609607_other3  Press release\nPORR hands over the Gleisarena ...

In [14]:
dfl1=pd.concat([dfl["File_name"].str.strip(),dfl["Data"]],axis=1)
dfl1.head()

File_name                                               Data
0  AT0000609607_2020_earn  Press release\nPORR: Record order backlog prov...
1   AT0000609607_2020_pre  Press release\nPreliminary figures 2019:\nPORR...
2     AT0000609607_other1  Press Release\nPORR starts into 2020 with succ...
3     AT0000609607_other2  Press release\nPORR to modernise strategically...
4     AT0000609607_other3  Press release\nPORR hands over the Gleisarena ...

In [22]:
dfl1.shape

(15807, 2)

In [23]:
dfl1[dfl1["File_name"]=="AT0000609607_pre"]

Empty DataFrame
Columns: [File_name, Data]
Index: []

In [64]:
dfl.shape

(15807, 2)

In [65]:
dfl.shape,dfn.shape

((15807, 2), (12813, 3))

In [66]:
File_name_output3=[]
File_name3=[]
Data3=[]
for i in dfn["File_name_output"].str.strip():
    for j in dfl["File_name"].str.strip():
        if i==j:
            File_name_output3.append(i)
            File_name3.append(j)
            Data3.append(dfl[dfl["File_name"].str.strip()==i]["Data"].values[0])             #code for merging input and output including duplicate values
df0kld = pd.DataFrame({"File_name": File_name3,"File_name_output":File_name_output3,
                    "Data": Data3})
df0kld.shape

(12726, 3)

In [67]:
df0kld.head()

File_name     File_name_output  \
0  AT0000609607_other1  AT0000609607_other1   
1  AT0000609607_other2  AT0000609607_other2   
2  AT0000609607_other3  AT0000609607_other3   
3    AT0000641352_earn    AT0000641352_earn   
4  AT0000641352_other1  AT0000641352_other1   

                                                Data  
0  Press Release\nPORR starts into 2020 with succ...  
1  Press release\nPORR to modernise strategically...  
2  Press release\nPORR hands over the Gleisarena ...  
3  Press | 25.03.2020 | 20:00\n\nCA Immo achieves...  
4  Press | 01/22/2020 | 17:59\n\nTree work in the...

In [68]:
Final=df0kld[["File_name","Data"]]
Final.shape

(12726, 2)

In [76]:
Final.to_csv("Final_GF_12726_2020.csv", index=False)

In [77]:
df0kld["File_name_output"].value_counts()       #No duplicate values are present

AT0000609607_other1     1
IT0003027817_other17    1
IT0003027817_other19    1
IT0003056386_Q1         1
IT0003057624_earn       1
                       ..
FR0004183960_earn       1
FR0004183960_other1     1
FR0004186856_earn       1
FR0004186856_Q1         1
SE0012674232_other2     1
Name: File_name_output, Length: 12726, dtype: int64

In [27]:
print("Unique: ",len(dfn["File_name_output"].unique()),",","Total: ",len(dfn["File_name_output"])) #Expected

Unique:  12813 , Total:  12813


In [70]:
print("Unique: ",len(df0kld["File_name_output"].unique()),",","Total: ",len(df0kld["File_name_output"])) #Actual

Unique:  12726 , Total:  12726


In [69]:
print("Unique: ",len(dfl["File_name"].unique()),",","Total: ",len(dfl["File_name"])) #Total

Unique:  15807 , Total:  15807


In [71]:
len(set(dfl["File_name"].str.strip())&set(dfn["File_name_output"].str.strip()))    #Common values

12726

In [72]:
len(dfl["File_name"].str.strip().unique())   #No duplicates in original data

15807

In [87]:
15807-12813,12813-12726

(2994, 87)

In [83]:
Rows_3081=list(set(dfl["File_name"].str.strip().unique())-set(dfn["File_name_output"].str.strip().unique()))
len(Rows_3081)                      #Total-output

3081

In [85]:
Rows_3081[:7]   #total should not be in output

['FR0004125920_earn',
 'FR0000125007_other8',
 'GB00BJVQC708_other12',
 'GB00B23JN426_other3',
 'CH0432492467_Q1',
 'GB00BTFR4F17_other1',
 'NO0010657448_other14']

In [16]:
Rows_87a=list(set(dfn["File_name"].unique())-set(dfl1["File_name"].unique()))
len(Rows_87a)

87

In [20]:
pd.Series(Rows_87a).to_csv("No_data_87_2020.csv",index=False)

In [79]:
Rows_87=list(set(dfn["File_name_output"].str.strip().unique())-set(dfl["File_name"].str.strip().unique()))
len(Rows_87)   #output-total

87

In [80]:
Rows_87[:8]  #output not in total

['GB0002570660_other14',
 'BE0003826436_Q1',
 'FI0009000400_other5',
 'GB00B0358H47_other15',
 'DE000A0Z23G6_earn',
 'NO0010833262_other8',
 'GB0006776081_other14',
 'FI0009000400_other11']

In [59]:
dfl[dfl["File_name"].str.strip()=="GB00BZ58DH10_other1"] #output not in total

Empty DataFrame
Columns: [File_name, Data]
Index: []

In [82]:
dfl[dfl["File_name"].str.strip()=="GB0002570660_other14"]

Empty DataFrame
Columns: [File_name, Data]
Index: []

In [81]:
dfn[dfn["File_name_output"].str.strip()=="GB0002570660_other14"] #total not in output

isin disclosure      File_name_output
5109  GB0002570660    other14  GB0002570660_other14

In [86]:
dfn[dfn["File_name_output"].str.strip()=="FR0004125920_earn"]#total not in output

Empty DataFrame
Columns: [isin, disclosure, File_name_output]
Index: []

Conclusion: Unique Total input files-Output: 15807-12813=2994 Matching Total&Output: 12726, No data : 12813-12726=87(Need to get from client) Not present in output file: 3081(Perhaps analysis is not necessary)


In [90]:
New_missing_data_87=list(set(dfn["File_name_output"].str.strip().unique())-set(df0kld["File_name_output"].str.strip().unique()))
New_missing_data_87[:5]

['GB0002570660_other14',
 'BE0003826436_Q1',
 'FI0009000400_other5',
 'GB00B0358H47_other15',
 'DE000A0Z23G6_earn']

In [89]:
len(New_missing_data_87)

87

In [93]:
Rows=set(Rows_3081)&set(New_missing_data_87)
len(Rows)                                                #No common in both

0

In [98]:
dfL=pd.read_excel(r'D:\GF\reshape_JanMay2020.xlsx', sheet_name="Litigious")
dfL.head()

isin  country               exchange    bvd_sector   sic  comname  \
0  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
1  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
2  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
3  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   
4  AT0000609607  Austria  Vienna Stock Exchange  Construction  1521  PORR AG   

  disclosure       date  ABOVEMENTIONED  ABROGATE  ...  WHOMSOEVER  WHOSOEVER  \
0        pre 2020-03-10             NaN       NaN  ...         NaN        NaN   
1       earn 2020-04-27             NaN       NaN  ...         NaN        NaN   
2     other1 2020-01-31             NaN       NaN  ...         NaN        NaN   
3     other2 2020-02-05             NaN       NaN  ...         NaN        NaN   
4     other3 2020-04-02             NaN       NaN  ...         NaN        NaN   

   WILFUL  WILLFUL  WILLFULLY  WILLFULNESS  WITNESS  WITNESSES  WRIT  WRITS  
0     NaN      NaN        NaN          NaN      NaN        NaN   NaN    NaN  
1     NaN      NaN        NaN          NaN      NaN        NaN   NaN    NaN  
2     NaN      NaN        NaN          NaN      NaN        NaN   NaN    NaN  
3     NaN      NaN        NaN          NaN      NaN        NaN   NaN    NaN  
4     NaN      NaN        NaN          NaN      NaN        NaN   NaN    NaN  

[5 rows x 912 columns]

In [99]:
dfL.shape

(12813, 912)